src/Http/ResponseBuilder.php

```diff
@@ -54,15 +69,19 @@ public function __construct($responseFactory)
      *
      * @param  int|null $statusCode
      * @param  array    $headers
+     * @param  bool     $successFlag
      * @return \Illuminate\Http\JsonResponse
      */
-    public function respond(int $statusCode = null, array $headers = []):JsonResponse
+    public function respond(int $statusCode = null, array $headers = [], bool $successFlag = true):JsonResponse
```

Comment:
Seeing as ResponseBuilder is an abstract class, you must use the success or error builder respectively. I feel like it's a bit strange being able to call `$successResponseBuilder->respond(200, [], false)`, it both is and isn't a success :p 

Instead of overriding the `respond` method, we could also add a protected abstract method `isSuccessResponse` which must return a boolean and be implemented by any child classes.